tests/test_coordinate_source.py

```diff
@@ -0,0 +1,328 @@
+                    # .coord file usually contains just the centers
+                    coord.write(f"{center[0]}\t{center[1]}\n")
+
+            # populate star file with particle centers
+            x_coords = [center[0] for center in _centers]
+            y_coords = [center[1] for center in _centers]
+            blocks = OrderedDict(
+                {
+                    "coordinates": DataFrame(
+                        {"_rlnCoordinateX": x_coords, "_rlnCoordinateY": y_coords}
+                    )
+                }
+            )
+            starfile = StarFile(blocks=blocks)
+            starfile.write(self.star_fp)
+
+            # populate box file with coordinates in EMAN1 .box format
+            with open(self.box_fp, "w") as box:
+                for center in _centers:
+                    # to make a box file, we convert the centers to lower left
+                    # corners by subtracting half the particle size (here: 256)
+                    lower_left_corners = (center[0] - 128, center[1] - 128)
+                    box.write(
+                        f"{lower_left_corners[0]}\t{lower_left_corners[1]}\t256\t256\n"
+                    )
+            # populate the second box file with nonsquare coordinates
+            with open(self.box_fp_nonsquare, "w") as box_nonsquare:
+                for center in _centers:
+                    # make a bad box file with non square particles
+                    lower_left_corners = (center[0] - 128, center[1] - 128)
+                    box_nonsquare.write(
+                        f"{lower_left_corners[0]}\t{lower_left_corners[1]}\t256\t100\n"
+                    )
+
+        # Now construct a simulated Relion directory
+        # to test RelionCoordinateSource
+        # The autopick.star file points to sample1.mrc
+        # (created earlier) and relion_coord.star as the coord file
+        os.makedirs(os.path.join(self.data_folder, "AutoPick/job006/"))
+        self.autopick_star_path = os.path.join(
+            self.data_folder, "AutoPick/job006/sample_relion_autopick.star"
+        )
+        shutil.copyfile(
+            os.path.join(self.test_dir_root, "sample_relion_autopick.star"),
+            self.autopick_star_path,
+        )
+        shutil.copyfile(
+            os.path.join(self.test_dir_root, "sample_relion_coord.star"),
+            # renamed because later we glob for sample*star and we don't want this one
+            os.path.join(self.data_folder, "relion_coord.star"),
+        )
+
+        # create default object from a .box file, for comparisons in tests
+        # also provides an example of how one might use this in a script
+        self.all_mrc_paths = sorted(glob(self.data_folder + "/*.mrc"))
+        self.all_box_paths = sorted(glob(self.data_folder + "/sample*.box"))
+        self.files_box = list(zip(self.all_mrc_paths, self.all_box_paths))
+        self.src_from_box = EmanCoordinateSource(self.files_box)
```

Comment:
Shouldn't this be in `testLoadFromBox`? If this crashes, we'd like it to be in a test, not in setup. We can regenerate it later on for other tests if needed.

Reply:
Good point thank you